core/src/test/java/org/bptlab/cepta/sum-of-delay-at-station/SumOfDelayAtStationTests.java

```diff
@@ -59,4 +65,37 @@ public void TestSumOfDelaysAtStation() throws IOException {
         
         Assert.assertTrue(pass);
     }
+
+    @Test
+    public void TestSumOfDelaysAtStationWithLiveTrainData() throws IOException {
+
+        boolean pass = true;
+        Long expectedStation1 = 1L;
+        Double expectedDelayAtStation1 = 3.0;
+
+        SumOfDelayAtStationFunction sumOfDelayAtStationFunction = new SumOfDelayAtStationFunction<LiveTrainData>();
+        DataStream<LiveTrainData> delayNotificationStream = LiveTrainDataProvider.liveTrainDatStreamWithDuplicates();
+
+        DataStream<Tuple2<Long, Double>> locationAndDelayStream = sumOfDelayAtStationFunction.SumOfDelayAtStation(delayNotificationStream, 3, "StationId");
+        ArrayList<Tuple2<Long, Double>> locationAndDelayArray = new ArrayList<>();
+        Iterator<Tuple2<Long, Double>> iterator = DataStreamUtils.collect(locationAndDelayStream);
+        while(iterator.hasNext()){
+            Tuple2<Long, Double> tuple = iterator.next();
+            locationAndDelayArray.add(tuple);
+        }
+        // check if any tuple is present
+        if (locationAndDelayArray.size() == 0) {
+            pass = false;
+        }
+
+        for (Tuple2<Long, Double> tuple : locationAndDelayArray) {
+            // check if first station matches expected delay
+            if (tuple.f0.equals(expectedStation1)) {
+                if (!tuple.f1.equals(expectedDelayAtStation1)) {
+                    pass = false;
+                }
+            } 
+        }
```

Comment:
How many entries do we expect here? If its just one, since we only input 3 events with, i suspect we only get one tuple as our output, so imo we shouldnt loop over there

Reply:
maybe it could be more if the function wasn't working right so i kept the loop